src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java

```diff
@@ -8,45 +8,145 @@
   @Test(groups = "mercury_search_cancelSearchPhrase")
   public void mercury_search_clearSearchPhrase() {
-    String searchingPhrase = "Infobox";
-    init();
+    String currentPhrase = new ArticlePage()
+        .open(MercurySubpages.MAIN_PAGE)
+        .getTopBar()
+        .openSearch()
+        .typeInSearch(this.SEARCH_PHRASE)
+        .clickClearSearchButton()
+        .getSearchPhrase();
+
+    Assertion.assertTrue(currentPhrase.isEmpty());
+  }
+
+  @Test(groups = "mercury_search_verifySearchLayout")
+  public void mercury_search_verifySearchLayout() {
+    Search search =
+        new ArticlePage()
+            .open(MercurySubpages.MAIN_PAGE)
+            .getTopBar()
+            .openSearch();
+
+    Assertion.assertTrue(search.isSearchInputFieldVisible());
+    Assertion.assertTrue(search.isClearSearchButtonVisible());
+    Assertion.assertTrue(search.isInputFieldSearchIconVisible());
+  }
+
+  @Test(groups = "mercury_search_userIsRedirectedToSearchResultsPage")
+  public void mercury_search_userIsRedirectedToSearchResultsPage() {
+    SearchResultsPage searchResults =
+        new ArticlePage()
+            .open(MercurySubpages.MAIN_PAGE)
+            .getTopBar()
+            .openSearch()
+            .typeInSearch(this.SEARCH_PHRASE)
+            .clickEnterAndNavigateToSearchResults();
+
+    Assertion.assertTrue(searchResults.isSearchResultsPageOpen());
+  }
+
+  @Test(groups = "mercury_search_searchResultsPageHasNoSearchIconInTopBar")
+  public void mercury_search_searchResultsPageHasNoSearchIconInTopBar() {
+    SearchResultsPage resultsPage =
+        new SearchResultsPage()
+            .openForQuery(SEARCH_PHRASE);
+
+    Assertion.assertFalse(resultsPage.getTopBar().isSearchIconClickable());
+  }
+
+  @Test(groups = "mercury_search_searchInputDoesNotCoverNavigation")
+  public void mercury_search_searchInputDoesNotCoverNavigation() {
+    SearchResultsPage resultsPage =
+        new SearchResultsPage()
+            .openForQuery(SEARCH_PHRASE);
+
+    resultsPage
+        .getTopBar()
+        .openNavigation();
+
+    Assertion.assertFalse(resultsPage.getSearch().isSearchInputFieldEditable());
+  }
+
+  @Test(groups = "mercury_search_searchNoResultsPageDisplayed")
+  public void mercury_search_searchNoResultsPageDisplayed() {
+    SearchResultsPage searchResults =
+        new SearchResultsPage()
+            .openForQuery(SEARCH_PHRASE_NO_RESULTS);
+
+    Assertion.assertTrue(searchResults.isNoResultsPagePresent());
+  }
+
+  @Test(groups = "mercury_search_noSuggestionsOnSearchResultsPage")
+  public void mercury_search_noSuggestionsOnSearchResultsPage() {
+    SearchResultsPage searchResults =
+        new SearchResultsPage()
+            .openForQuery(SEARCH_PHRASE_NO_RESULTS);
+
+    Assertion.assertFalse(searchResults.getSearch().areSearchSuggestionsDisplayed());
+  }
+
+  @Test(groups = "mercury_search_focusOnTryAnotherSearchWhenNoResults")
+  public void mercury_search_focusOnTryAnotherSearchWhenNoResults() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)